core/state_transition.go

```diff
@@ -384,9 +389,57 @@ func (st *StateTransition) StakingTransitionDb() (usedGas uint64, err error) {
 	return st.gasUsed(), err
 }
 
+func (st *StateTransition) checkDuplicateFields(validator common.Address, name, identity string, blsKeys []shard.BLSPublicKey) error {
+	addrs, err := st.bc.ReadValidatorList()
+	if err != nil {
+		return err
+	}
+
+	checkName := name != ""
+	checkIdentity := identity != ""
+	checkBlsKeys := len(blsKeys) != 0
+
+	blsKeyMap := map[shard.BLSPublicKey]struct{}{}
+	for _, key := range blsKeys {
+		blsKeyMap[key] = struct{}{}
+	}
+
+	for _, addr := range addrs {
+		if bytes.Compare(validator.Bytes(), addr.Bytes()) != 0 {
+			wrapper, err := st.state.ValidatorWrapperCopy(addr)
+
+			if err != nil {
+				return err
+			}
+
+			if checkName && wrapper.Name == name {
```

Comment:
Why cant the validators have same name and identity? Do they have to be unique? Blskey wise, how can there be duplicates when bls key is generated by the protocol amd unique per validator?

Reply:
@denniswon we have issues with node operator mistakenly re-using the same BLS for a different validator

Reply:
regarding the name and identify this is more to avoid people trying to impersonate another validator, though they could still trick a delegator with some character difference : "P-OPS validator| vs "P-0PS validator"

Reply:
but I am not sure if we should make identity and name as unique at the protocol level. or should do at the sdk/cli level. idk, not sure if having more than bls key being unique to validator, other info do not necessarily have to be unique at the protocol level. _shrug_